packages/drivers/odsp-driver/src/createFile.ts

```diff
@@ -81,14 +81,10 @@ export async function createNewFluidFile(
     if (createNewSummary !== undefined && createNewCaching) {
         assert(summaryHandle !== undefined, "Summary handle is undefined");
         // converting summary and getting sequence number
-        const snapshot: IOdspSnapshot = convertCreateNewSummaryTreeToIOdspSnapshot(createNewSummary, summaryHandle);
-        const protocolSummary = createNewSummary.tree[".protocol"] as ISummaryTree;
-        const documentAttributes = getDocAttributesFromProtocolSummary(protocolSummary);
-        const sequenceNumber = documentAttributes.sequenceNumber;
-
+        const snapshot: ISnapshotValue = convertCreateNewSummaryTreeToTreeAndBlobs(createNewSummary, summaryHandle);
         // caching the converted summary
-        const value: ISnapshotCacheValue = { snapshot, sequenceNumber };
-        await epochTracker.put(createCacheSnapshotKey(odspResolvedUrl), value);
+        const cacheValue: ISnapshotCacheValueWithVersion = { ...snapshot, version: 2 };
```

Comment:
EpochTracker.put():
```typescript
    public async put(entry: IEntry, value: any) {
        assert(this._fluidEpoch !== undefined, 0x1dd /* "no epoch" */);
        const data: IVersionedValueWithEpoch = {
            value,
            version: persistedCacheValueVersion,
            fluidEpoch: this._fluidEpoch,
        };
```

Which means this code should not deal with versions.
Is this change intentional?

Also version of persistedCacheValueVersion should be bumped, as content takes different form, right?

Reply:
"persistedCacheValueVersion" deals with cache data version and is for anything put in the cache. The SnapshotCacheValue Version represents version of snapshot data as not we can have different formats. So the host cache can contain cached snapshot and needs to be treated as such while this newer binary snapshot stored with version tells us what kind of snapshot it  is.

Reply:
ISnapshotCacheValueWithVersion just represents the value in   const data: IVersionedValueWithEpoch = {
            value,
            version: persistedCacheValueVersion,
            fluidEpoch: this._fluidEpoch,
        };

Reply:
I'd simplify and only support new format going forward. I.e. no matter what wire format is used, it is always converted to ISnapshotValue and that is what stored in cache and assumed to be in the cache. persistedCacheValueVersion is bumped to start ignoring old entries.
I think it's much simpler model then adding version on top of version and complicating both read & write code.

Reply:
Basically persistedCacheValueVersion's goal is precisely to deal with changes in caching schema format by ignoring old entries - there is some perf loss of not using data in old format, but it's very short lived, we should optimize toward simplicity...

Reply:
For newer entries I am storing the ISnapshotValue only. I was adding version for figuring out old entries as I missed we can do that with persistedCacheValueVersion. So will remove extra version.